src/praisonai-agents/praisonaiagents/gateway/protocols.py

```
@@ -2298,6 +2299,125 @@ def drain_requested(
+            the directory containing this module's package.
+
+    Returns:
+        An opaque, non-secret fingerprint string, or ``None`` if it cannot be
+        determined.
+    """
+    import os
+
+    if checkout_dir is None:
+        try:
+            checkout_dir = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+        except Exception:
```

Ruff (0.15.18)

[warning] Do not catch blind exception: `Exception`

(BLE001)